### DISCUSSION TITLE
How can I insert a small geometry in a larger one?

### DISCUSSION THREAD

**Discussion by kstcmcjw** [nrc-cnrc/EGSnrc#883] [Questions] [answered]
Hi, every EGSnrc guider, forgive me of my stupid, I faced a realistic problem, for example, in the air box, i want to insert a chamber in the special site in the air box, this problem seems not a EGS_cdgeometry can slove this problem, as the EGS_cdgeometry is the "cutoff" plane and combination plane, it seems that not suitable for the "insert" chamber?
I have tried with the  EGS_cdgeometry, defined the air box and the chamber, but the EGS_view only display the air box, neglect the chamber, but if i remove the air box, the chamber displayed nomally. So i think whether the EGS_library i have choosed was not right?

That is my aim:
![image](https://user-images.githubusercontent.com/96934217/175845512-8d49652a-37c5-4e65-a67b-5a819fd473c7.png)

And this is my program as the follow:

###2.GEOMETRY
:start geometry definition:
  :start geometry: 
    name = cd_planes
    library = egs_planes
    type = EGS_Zplanes
    positions = -80 -24.2 -0.2 0 500
  :stop geometry:

  :start geometry:
    name = kongqi
    library = egs_box
    box size =  600 600 600
        :start media input:
            media = kongqi
        :stop media input:
  :stop geometry:
  simulation geometry = kongqi

  :start geometry:
    name = barrier_lead
    library = egs_box
    box size = 100 100 100
        :start media input:
            media = lead
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_lead

 :start geometry:
    name = barrier_concrete
    library = egs_box
    box size = 100 100 100
        :start media input:
            media = concrete
        :stop media input:
  :stop geometry:
  simulation geometry = barrier_concrete

  :start geometry:
    name = chamber-kongqi
    library = egs_spheres
    midpoint = 0 0 -54.2
    radii = 0.6204 0.7204
    :start media input:
        media = kongqi glass
        set medium = 1 1
        set medium = 2 2
    :stop media input:
:stop geometry:
simulation geometry = chamber-kongqi
  
  :start geometry:
    name = protection
    library = egs_cdgeometry
    base geometry =cd_planes
    set geometry = 3 chamber-kongqi
    set geometry = 3 kongqi
    set geometry = 2 barrier_lead
    set geometry = 1 barrier_concrete
    set geometry = 0 kongqi

  :stop geometry:
  simulation geometry = protection

:stop geometry definition:

Thanks for your attention, have a nice day!

**Comment by kstcmcjw** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/96934217/175871373-2661a089-1861-4e09-a2f3-12cfe59473ff.png)

It seems i find the answer!  Is it right?

**Reply by mxxo** [reply]
That's right, envelope geometries (e.g. `egs_genvelope`) are the way to do this.

**Reply by kstcmcjw** [reply]
> 

Thank you very much, I get it just now!

![image](https://user-images.githubusercontent.com/96934217/175938049-dd892650-f926-4d81-be0f-410dae8bf79c.png)

Then the chamber said "I am here, I come out at last! ". Haha!
